thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/DetectionConfigTuner.java

```diff
@@ -103,12 +99,13 @@ public DetectionConfigTuner(DetectionConfigDTO config, DataProvider dataProvider
     Tunable tunable = instantiateTunable(componentClassName, yamlParams, dataFetcher);
 
     // round to daily boundary
-    String metricName = componentProps.get(PROP_METRIC).toString();
-    String datasetName = componentProps.get(PROP_DATASET).toString();
-    MetricConfigDTO metricConfig = dataProvider.fetchMetric(metricName, datasetName);
-    DatasetConfigDTO datasetConfig = dataProvider.fetchDatasets(Collections.singletonList(metricConfig.getDataset()))
-        .get(metricConfig.getDataset());
-    String metricUrn = MetricEntity.fromMetric(ConfigUtils.getMap(componentProps.get(PROP_FILTERS)), metricConfig.getId()).getUrn();
+    Preconditions.checkNotNull(componentProps.get(PROP_METRIC_URN));
+    String metricUrn = componentProps.get(PROP_METRIC_URN).toString();
+    MetricEntity metricEntity = MetricEntity.fromURN(metricUrn);
+    Map<Long, MetricConfigDTO> metricConfigMap = dataProvider.fetchMetrics(Collections.singleton(metricEntity.getId()));
+    Preconditions.checkArgument(metricConfigMap.size() == 1, "Unable to find the metric to tune");
+
+    DatasetConfigDTO datasetConfig = metricConfigMap.values().iterator().next().getDatasetConfig();
```

Comment:
In my memory, ` metricConfig.getDatasetConfig()` won't return the dataset config due to legacy problems. Have to do
```
dataProvider.fetchDatasets(Collection.signleton(metric.getDataset()))
```